docs/extended_api/pipelines/pipeline_role.md

```diff
@@ -0,0 +1,37 @@
+---
+grand_parent: Extended API
+parent: Pipelines
+---
+
+# `cuda::pipeline_role`
+
+```c++
+// Header `<cuda/pipeline>`.
```

Comment:
Same comment about "declared in" style as earlier.

This file needs a comment that this functionality is not a part of the github-released open-source library.